Bottom: 3df0369a72987bf3ee81dfddd55c18c542bcf024
Top:    b9284e0657a83b63abf7d88a6b923a0baf54feb5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2017-03-17 23:18:27 +0000

action.c Remove file scope static variable element_cache

We don't tend to load the same refdes multiple times, and don't actually
even use the cache in the _possibly_ benefical case of looking up elements
repeatedly when setting multiple attributes on them.


---

diff --git a/src/action.c b/src/action.c
index 5c05c1d..331d0b7 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7015,23 +7015,15 @@ ActionPSCalib (int argc, char **argv, Coord x, Coord y)
 
 /* --------------------------------------------------------------------------- */
 
-static ElementType *element_cache = NULL;
-
 static ElementType *
 find_element_by_refdes (char *refdes)
 {
-  if (element_cache
-      && NAMEONPCB_NAME(element_cache)
-      && strcmp (NAMEONPCB_NAME(element_cache), refdes) == 0)
-    return element_cache;
-
   ELEMENT_LOOP (PCB->Data);
   {
     if (NAMEONPCB_NAME(element)
 	&& strcmp (NAMEONPCB_NAME(element), refdes) == 0)
       {
-	element_cache = element;
-	return element_cache;
+        return element;
       }
   }
   END_LOOP;
@@ -7127,7 +7119,6 @@ ActionElementList (int argc, char **argv, Coord x, Coord y)
 	CLEAR_FLAG (FOUNDFLAG, element);
       }
       END_LOOP;
-      element_cache = NULL;
       number_of_footprints_not_found = 0;
 
       /* Empty out the unplaced element list.
@@ -7287,7 +7278,6 @@ ActionElementList (int argc, char **argv, Coord x, Coord y)
 	SetChangedFlag (true);
 
       /* Now reload footprint */
-      element_cache = NULL;
       e = find_element_by_refdes (refdes);
       e_owner = PCB->Data;
     }
